### DISCUSSION TITLE
Default Midi Learn , and Incoming Midi Muted with inactive clip

### DISCUSSION THREAD

**Discussion by MasterStatus** [SynthstromAudible/DelugeFirmware#897] [Ideas]
Would love to see either a saveable midi learn into the default, or a device/port/channel as default, to avoid having to midi learn every time. Perhaps it auto links to the next midi channel number for the next and subsequent instruments created.

As an extension of this, the option to make a single midi source be 'learnt' to several clips/instruments (as is currently possible), but they only become active when the clip is active, or you are in that specific clip view. 

A strong benefit of a 'Incoming Midi being muted for un-active clips' feature would be that the Deluge could act as a patch selector for live play in Arranger mode i.e. you have one midi device linked to several deluge instruments clips or midi clips, and as the clip becomes active in the arranger timeline it turns it 'on', meaning 'hands-free' changing of instruments for your, for example, keyboard or midi drums.   This is preferable to Program Changes because PC messages are often not fast enough.

**Comment by seangoodvibes** [comment]
> Would love to see either a saveable midi learn into the default, or a device/port/channel as default, to avoid having to midi learn every time. Perhaps it auto links to the next midi channel number for the next and subsequent instruments created.
> 
> As an extension of this, the option to make a single midi source be 'learnt' to several clips/instruments (as is currently possible), but they only become active when the clip is active, or you are in that specific clip view.
> 
> A strong benefit of a 'Incoming Midi being muted for un-active clips' feature would be that the Deluge could act as a patch selector for live play in Arranger mode i.e. you have one midi device linked to several deluge instruments clips or midi clips, and as the clip becomes active in the arranger timeline it turns it 'on', meaning 'hands-free' changing of instruments for your, for example, keyboard or midi drums. This is preferable to Program Changes because PC messages are often not fast enough.

This is done and part of the nightly build. Please see the documentation on MIDI Follow Mode:

https://github.com/SynthstromAudible/DelugeFirmware/blob/community/docs/features/midi_follow_mode.md
